Edits on the IAM overview topic (#4950)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/concepts/iam-overview/index.md

- [Authorization factors](/docs/concepts/iam-overview-authorization-factors/): Describes design considerations for defining what resources that a user can access

- [Architectural factors](/docs/concepts/iam-overview-architectural-factors/): Describes important architectural requirements to consider for your solution and possible strategies for addressing them

- [IAM terminology](/docs/concepts/iam-overview-iam-terminology/): Contains key definitions of common terms

## IAM concepts

IAM secures your services in two ways. First, it verifies the identity of a user when they sign in. Second, it lets a user access only the parts of your network and services that they have permission for.

For example, a customer may require only a username and password and has permission to view and buy items. While a vendor may require a one-time passcode and has permission to add and update items, but not buy them.

### Workforce and Customer Identity

High-level designs for IAM may be called Workforce or Customer identity. The two have significant overlap in use cases and technical approaches. The most important thing is to design and build a solution that meets your particular requirements.

> - Workforce identity (Workforce or WF) solutions manage employee and contractor access to your organization's apps and resources. The main goal of WF solutions is to manage risk. IT teams usually assign user identities and use cases are typically administrative. For example, controlling access to apps by integrating them into your single sign-on solution.

Reviewer: FYI - these for examples need to be complete sentences.
```suggestion
- Workforce identity (Workforce or WF) solutions manage employee and contractor access to your organization's apps and resources. The main goal of WF solutions is to manage risk. IT teams usually assign user identities and use cases are typically administrative. For example, you might want to control access to apps by integrating them into your single sign-on solution.
```

> - Customer identity (Customer Identity and Access Management or CIAM) solutions add the ability to manage customer, partner, and other external access to WF. Ease of use for customers is important. CIAM solutions can increase both customer engagement and revenue. Unlike WF users, CIAM users commonly create their own identities, sign in from various locations and platforms, and may have multiple identities. Use cases are typically user-focused. For example, improving the user experience (UX) of an app and building branded user registration and sign-in flows.

Reviewer: ```suggestion
- Customer identity (Customer Identity and Access Management or CIAM) solutions add the ability to manage customer, partner, and other external access to WF. Ease of use for customers is important. CIAM solutions can increase both customer engagement and revenue. Unlike WF users, CIAM users commonly create their own identities, sign in from various locations and platforms, and may have multiple identities. Use cases are typically user-focused. For example, you might want to improve the user experience (UX) of an app and build branded user registration and sign-in flows.
```

#### Infrastructure

- **Scalability:** Design your solution to allow for user growth and changing use patterns of your apps and infrastructure, without requiring a redesign.

> - **Easy integration with app stack:** Maintain your IAM solution separately from your apps. Your apps can evolve independently and you can enable multiple apps to use your IAM solution. Enable integration with your apps that optimizes performance by designing an event-based interface that is accessed using an API. Provide an SDK to make integration easier.

Reviewer: ```suggestion
- **Easy integration with app stack:** Maintain your IAM solution separately from your apps. Your apps can evolve independently, and you can enable multiple apps to use your IAM solution. Enable integration with your apps that optimizes performance by designing an event-based interface that's accessed using an API. Provide an SDK to make integration easier.
```

> - **Traffic surge protection:** Prevent sudden bursts of requests from disrupting usage or operations.

Reviewer: avoid the use of "usage" haha - it's just jargon and very "markety"
```suggestion
- **Traffic surge protection:** Prevent sudden bursts of requests from disrupting use or operations.
```

- **High availability:** Ensure that your IAM solution is fully operational when your apps go live.

- **High reliability:** Ensure that your IAM solution is dependable and that every element of your solution works correctly, every time.

#### Security

- **User storage and password management:** Store and manage information about your users, including their passwords, for authenticating and authorizing users.

> - **Multifactor Authentication (MFA):** Require a user to verify their identity in different ways. Some of these ways include knowledge, such as answers to questions, or biometrics, such as a fingerprint, or possession, such as a key-card. MFA is key to controlling access to your apps.

Reviewer: Is there a better way that we can present this information in this sentence without using "such as" so much?
```suggestion
- **Multifactor Authentication (MFA):** Require a user to verify their identity in different ways. Some of these ways include knowledge, such as answers to questions, or biometrics, such as a fingerprint, or possession, such as a key-card. MFA is key to controlling access to your apps.
```

- **Automated onboarding:** Automate onboarding workflows and lifecycle management. For example, manage a new user's AWS SSO entitlements, capture document signatures, and provision and deprovision user app accounts. Automated onboarding minimizes the time and effort to provision new users and perform other lifecycle change procedures and creates a positive experience for new users.

- **Frictionless MFA experience:** Use special strategies, such as SSO or simplified sign-in flows from managed devices to streamline user sign-in flows.

## IAM design example

The following diagram shows a typical CIAM solution for an app that supports an organization's employees and customer users (B2B). End users sign in to the web portal or mobile app in various ways. All end users sign in to the same system, but sign-in behavior and available services are user- and organization-specific. The diagram shows some of IAM's complexity and areas where it affects an app's high-level design.

This is an AWS-hosted healthcare app but the IAM design features are generic and can apply to apps in any domain.

> The app is used by the organization's employees and by other large corporations, and has tens of millions of end users. Most end users are patients, and most primary users are physicians, nurses, and other medical personnel.

Reviewer: ```suggestion
The app is used by the organization's employees and by other large corporations and has tens of millions of end users. Most end users are patients, and most primary users are physicians, nurses, and other medical personnel.
```
